File: Assignment43/program3.c
//////////////////////////////////////////////////////////////////////
// File name :       program3.c
// Discription :     It is Recusive program to count small character in string
// Author :          Mangesh Ashok Bedre
// Date :            17/11/2025
//
//////////////////////////////////////////////////////////////////////

#include<stdio.h>

//////////////////////////////////////////////////////////////////////
// Function Name :   Small
// Discription :     It is used to count small character in given string
// Input  :          char *
// Output :          int
// Author :          Mangesh Ashok Bedre
// Date :            17/11/2025
//
//////////////////////////////////////////////////////////////////////

int Small(char * str)
{
    static int iCount = 0; 
    if(*str =='\0')
    {
        return iCount;
    }
    else
    {
        if(*str >= 'a' && *str <= 'z')
        {
            iCount++;
        }
        str++;
        Small(str);
    }
} 

int main()
{
    char ch[50] = {'\0'};
    int iRet = 0;

    printf("Enter the string : \n");
    scanf("%[^\n]s",ch);

    iRet = Small(ch);
    printf("Small characters are in string : %d\n",iRet);

    return 0;
}

File: Assignment55/program3.java
// Write Java program which accept array of characters from user and
// accept one character. Return occurrence of that character without
// considering case.
// Input : string : c B b u E B r b
//         char :  b
// Output : 4


import java.util.Scanner;

class MyArray
{
    public int Search(String str, char ch)
    {
        int iCount = 0,i = 0;

        ch = Character.toLowerCase(ch);
        str = str.toLowerCase();

        for(i = 0; i < str.length(); i++)
        {
            if(str.charAt(i) == ch)
            {
                iCount++;
            }
        }
        return iCount;
    }
}

class program3
{
    public static void main(String a[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the string : ");
        String str = sobj.nextLine();

        System.out.println("Enter the charater : ");
        char ch = sobj.next().charAt(0);

        MyArray mobj = new MyArray();
        int iRet = mobj.Search(str,ch); 

        System.out.println("Occurance of "+ ch+" in String are : " + iRet);
    }
}